torch/nn/functional.py

```diff
@@ -2130,6 +2130,9 @@ def binary_cross_entropy_with_logits(input, target, weight=None, size_average=No
     if not (target.size() == input.size()):
         raise ValueError("Target size ({}) must be the same as input size ({})".format(target.size(), input.size()))
 
+    if pos_weight is not None and not (pos_weight.size(-1) == input.size(-1)):
+        raise ValueError("pos_weight length ({}) must be the same as the number of classes ({})".format(pos_weight.size(-1), input.size(-1)))
```

Comment:
nit: it might be better to be neutral about what matches what: "The length of pos_weight ({}) and the number of classes (input.size(-1) = {}) should be the same". In the original issue, I think the user believes he/she has two classes and instead has a malformed input.